tests/qemu/qemu.pm

```diff
@@ -15,19 +15,31 @@ use warnings;
 use base "consoletest";
 use testapi;
 use utils;
+use transactional qw(trup_call check_reboot_changes);
+use Utils::Architectures qw(is_x86_64 is_ppc64le is_s390x is_aarch64);
+use version_utils qw(is_sle_micro);
 
+sub is_qemu_preinstalled {
+    return is_sle_micro && check_var('FLAVOR', 'MicroOS-Image') && !script_run 'rpm -q patterns-microos-kvm_host';
+}
+
+sub install_qemu {
+    my $qpkg = shift;
+    if (zypper_call("in $qpkg", exitcode => [0, 5])) {
```

Comment:
maybe better here to use `is_transactional` but you need also to modify that function to include SLE Micro distri.

Reply:
I have noticed that as well, or at least was not completely sure. Nevertheless, I find the feature of `zypper` that it informs users about transactional-server helpful. Maybe later on we can add this feature to `zypper_call` directly, so we won't need to update other testcases using `zypper_call` (and another motivation for other test case writers to use directly `zypper_call` instead of script_run derivates). 
I also see it is quite safe usage as we have 3 possible outcomes:
1) 0 -> success
2) 5 -> has specific fail handler
3) every other code is treated as a test case failure.

Reply:
Ok, if error `5` refers to "error due transactional system" I'm ok.

Reply:
http://kepler.suse.cz/tests/6154#step/user/2

```
       5 - ZYPPER_EXIT_ERR_PRIVILEGES
           User invoking zypper has insufficient privileges for specified operation.
```

Right, could happen with normal user trying to pull a package... I have missed that, sorry!

Just a quick idea how to harden it 
```
if (zypper_call("in $qpkg", exitcode => [0, 5]) && is_transactional) { ... }
```

Reply:
interesting, so the error code we get is cause it's RO, which could be in non-transactional systems... 
I would go with just the `is_transactional` check to make things simpler, but up to you.